From 8e9048053b57eb86672c83727283392bf139ed2c Mon Sep 17 00:00:00 2001
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Wed, 10 Jun 2015 08:38:22 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: program Func/DBI reg for cmd mode

When changing the sequence for video to send OTP commands in
DBI mode, programming of DSI Func and DBI register for command
mode got missed out which was present in the code already.
Bringing back the DSI Func/DBI register programming for
cmd mode in this patch.

This patch should be squashed into:
    commit f2aa9a4c66e371b6523e08702dae687278413fe1
    Author: Yogesh <yogesh.mohan.marimuthu@intel.com>
    Date:   Fri Apr 10 15:40:18 2015 +0530

            FOR_UPSTREAM [VPG]: drm/i915: send otp packets

Change-Id: I0a446a7a85c11a287171aba7be85dd815a32b6d9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7482
Signed-off-by: Yogesh <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 682e031..d6e5388 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -468,9 +468,22 @@ static void intel_dsi_pre_enable(struct intel_encoder *encoder)
 
 	do {
 		val = intel_dsi->lane_count << DATA_LANES_PRG_REG_SHIFT;
-		val |= intel_dsi->channel << VID_MODE_CHANNEL_NUMBER_SHIFT;
-		val |= intel_dsi->pixel_format;
+
+		if (is_cmd_mode(intel_dsi)) {
+			val |= intel_dsi->channel <<
+					CMD_MODE_CHANNEL_NUMBER_SHIFT;
+			val |= CMD_MODE_DATA_WIDTH_OPTION2;
+
+			I915_WRITE(MIPI_DBI_FIFO_THROTTLE(pipe),
+					DBI_FIFO_EMPTY_QUARTER);
+			I915_WRITE(MIPI_HS_LP_DBI_ENABLE(pipe), 0);
+		} else {
+			val |= intel_dsi->channel <<
+						VID_MODE_CHANNEL_NUMBER_SHIFT;
+			val |= intel_dsi->pixel_format;
+		}
 		I915_WRITE(MIPI_DSI_FUNC_PRG(pipe), val);
+
 		val = 0;
 		if (intel_dsi->eotp_pkt == 0)
 			val |= EOT_DISABLE;
-- 
1.9.1

